Indexing/CodeTree.cpp

```diff
@@ -666,25 +657,20 @@ std::ostream& operator<<(std::ostream& out, const CodeTree& ct)
 
 //////////////// insertion ////////////////////
 
-void CodeTree::CompileContext::init()
-{
-  nextVarNum=0;
-  varMap.reset();
-  nextGlobalVarNum=0;
-  globalVarMap.reset();
-}
+template<bool forLits>
+CodeTree::Compiler<forLits>::Compiler(CodeStack& code) : code(code), nextVarNum(0), nextGlobalVarNum(0) {}
 
-void CodeTree::CompileContext::nextLit()
+template<bool forLits>
+void CodeTree::Compiler<forLits>::nextLit()
 {
-  nextVarNum=0;
+  ASS(forLits);
```

Comment:
Is it worth templating just some of the methods in `Compiler` rather than the whole class to avoid this weirdness? I'm not sure, just wondering.

Reply:
You mean the entire `CodeTree`? Well, there is the `bool _clauseCodeTree` member which I want to convert to a template flag in the future, and that would render templating `Compiler` unnecessary. I can try to do this now, not sure though how much would be changed.

Reply:
Just the `Compiler` - but I tried and it sucked so forget I said anything!

Reply:
I'm sorry, what did you try? 😃

Reply:
Tried removing the template parameter from the `Compiler` class and inserting it for the methods that need it, but if anything it's more error-prone.